uk.co.compucorp.civicrm.hrcore/CRM/HRCore/Upgrader/Steps/1032.php

```diff
@@ -0,0 +1,140 @@
+<?php
+
+trait CRM_HRCore_Upgrader_Steps_1032 {
+
+  /**
+   * Deletes some demographics fields where they are not in use
+   *
+   * @return bool
+   */
+  public function upgrade_1032() {
+
+    $this->up1032_modifyNewIndividualMenu();
+    $this->up1032_createStaffDirectoryMenu();
+    $this->up1032_createComunicationsMenu();
+    $this->up1032_moveNewEmailSubmenu();
+    $this->up1032_createComunicationsSubmenus();
+    $this->up1032_removeOldMenuItems([
+      'New Organisation',
+      'New Organization',
+      'New Group',
+      'Manage Groups',
+      'Find and Merge Duplicate Contacts',
+    ]);
+    $this->up1032_disableImportMenus([
+      'Import Contacts',
+      'Import / Export',
+      'Import Custom Fields',
+    ]);
+
+    return TRUE;
+  }
+
+  private function up1032_modifyNewIndividualMenu() {
+    civicrm_api3('Navigation', 'get', [
+      'label' => 'New Individual',
+      'api.Navigation.create' => [
+        'id' => '$value.id',
+        'label' => 'Add New Staff Member',
+        'name' => 'Add New Staff Member',
+        'url' => 'civicrm/import/contact?reset=1&force=1',
+      ],
+    ]);
+  }
+
+  private function up1032_createStaffDirectoryMenu() {
+    $menu = civicrm_api3('Navigation', 'get', [
+      'label' => 'Add New Staff Member',
+    ]);
+
+    $menu = array_shift($menu['values']);
+    civicrm_api3('Navigation', 'create', [
+      'label' => 'Staff Directory',
+      'name' => 'Staff Directory',
+      'url' => 'civicrm/contact/search/advanced?reset=1&force=1',
+      'parent_id' => $menu['parent_id'],
+      'domain_id' => $menu['domain_id'],
+      'permission' => $menu['permission'],
+      'is_active' => 1,
+      'weight' => 2,
+      'has_separator' => 1,
+    ]);
+    // If we don't flush it will not recognize newly created parent_id
+    CRM_Core_PseudoConstant::flush();
+  }
+
+  public function up1032_createComunicationsMenu() {
+    $menu = civicrm_api3('Navigation', 'get', [
+      'label' => 'Add New Staff Member',
+    ]);
+
+    $menu = array_shift($menu['values']);
+    civicrm_api3('Navigation', 'create', [
+      'label' => 'Record Communication',
+      'name' => 'Record Communication',
+      'parent_id' => $menu['parent_id'],
+      'domain_id' => $menu['domain_id'],
+      'permission' => $menu['permission'],
+      'is_active' => 1,
+      'weight' => 3,
+    ]);
+    // If we don't flush it will not recognize newly created parent_id
+    CRM_Core_PseudoConstant::flush();
+  }
+
+  private function up1032_moveNewEmailSubmenu() {
+    $menu = civicrm_api3('Navigation', 'get', [
+      'label' => 'Record Communication',
+    ]);
+
+    $menu = array_shift($menu['values']);
+    civicrm_api3('Navigation', 'get', [
+      'label' => 'New Email',
+      'api.Navigation.create' => [
+        'id' => '$value.id',
+        'parent_id' => $menu['id'],
+        'has_separator' => 0,
+        'weight' => 1,
+      ],
+    ]);
+  }
+
+  public function up1032_createComunicationsSubmenus() {
```

Comment:
Kindly change the access modifier, fix name typo and add comment as stated for the other menu above.